### DISCUSSION TITLE
Release cycle

### DISCUSSION THREAD

**Discussion by blinxen** [GitoxideLabs/gitoxide#912] [General]
How does the release cycle for all the gix-* crates look like? Are all crates released at the same time or is every crate released differently?

**Comment by Byron** [comment]
Crates are released "as needed" and need is determined exclusively by looking at the git history of itself and its dependencies. Conventional commit messages are used to indicate if something is breaking or not, and based on that, [`cargo-smart-release`](https://github.com/Byron/gitoxide/blob/main/cargo-smart-release/README.md) will determine which version to set and which crates to publish. It's worth noting that breaking changes propagate forward to all dependents to isolate downstream consumers from them.

This is why a breaking change in a crate like `gix-features` that is indirectly used by most other `gix-*` crates can cause all of them to be republished with a breaking change of their own.

I hope that helps.

**Reply by blinxen** [reply]
Thanks for the answer!
